text/0230-telemetry-viewer.md

```diff
@@ -0,0 +1,103 @@
+# Telemetry Viewer for Developers
+
+_A local telemetry viewer to aid in instrumentation and debugging of pipelines._
+
+## Motivation
+
+OpenTelemetry offers a rich, highly customizable, and highly configurable
+ecosystem of tooling, SDKs, APIs, and instrumentation libraries. However, with
+this complexity comes a cost -- the barrier to entry for new users can be very
+high, with significant cycle time required in order to understand how their
+instrumentation code changes affect the instrumentation emitted by their
+service.
+
+A 'local development' experience for OpenTelemetry would aid in reducing this
+cycle time and understandability burden from developers.
+
+## Explanation
+
+Different users of a telemetry system have different needs and expectations for
+debugging instrumentation. Currently, developers and coders have two options for
+quick feedback - using a logging exporter at the collector or SDK, or using an
+existing analysis tool (open source or proprietary). Both of these options have
+drawbacks - the logging exporter presents an overwhelming amount of text-based
+data, and depending the characteristics of a development environment, it may be
+challenging to stand up and use a local suite of open source analysis tools
+(such as Jaeger, Prometheus, and OpenSearch) or use a commercial tool.
+
+Reduced cycle time (both in a DevOps sense and also in a more general reading of
+the word) is a contributor to quality and resiliency of software and human
+systems. Being able to quickly get feedback about if your changes are having the
+desired effect or not is invaluable, especially for developers that are
+beginning to instrument their services for observability.
+
+The goal of this OTEP is to define a set of requirements for a solution to this
+problem. Ultimately, the vision is that a developer would be able to use a
+collector extension to view the following:
+
+- Metrics, Trace, and Log Data collected over the last X minutes.
+- The current configuration, pipelines, and operating metrics of the collector.
+- A list of instrumentation libraries, agents, or other ecosystem components in
+  use by the pipelines.
+- All attribute and resource keys seen by the collector over the last X minutes.
+
+## Internal details
+
+Broadly, the implementation of this viewer should be a collector extension that
+exposes a simple web portal for viewing data along with some sort of data store
+to hold the data emitted. This extension could be bundled with specific
+collector releases, or brought in via the collector builder.
```

Comment:
I am just thinking about the practical use: would the extension be part of one of the existing collector distributions (core, contrib) or would there be a separate distribution that provides something ready to use?

Also an enduser would still need to write a collector config? So they can not simple say `otel-desktop-viewer` in one window and then send telemetry via OTLP to that component. Is having a "default config" within scope?

Reply:
Added some clarifying comments